include/shark/Models/Trees/CARTClassifier.h

```diff
@@ -86,6 +86,15 @@ class CARTClassifier : public AbstractModel<RealVector,LabelType>
 			ar & r;
 			ar & g;
 		}
+		NodeInfo() {}
+		explicit NodeInfo(std::size_t nodeId) : nodeId(nodeId) {}
+		NodeInfo(std::size_t nodeId, LabelType const& label) : nodeId(nodeId), label(label) {}
+		NodeInfo(std::size_t nodeId, LabelType && label)
+				: nodeId(nodeId), label(std::move(label)) {}
+		NodeInfo(NodeInfo const&) = default;
+		NodeInfo(NodeInfo &&) = default;
```

Comment:
They won't be auto-generated so long as there are any declared constructors, however, you might still right that we don't need the two of them as `label' is likely small (<10 elements), and thus would not benefit from the move, if blas::vector implements the small-vector optimization. 

If there is no smal-vector optimization, I can also rewrite them using boost macros, so we still get the move support on supporting compilers?